<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="consensusNet provides tools for combining the results of
    metagene clustering in different studies, datasets, and using
    different methods.  It combines results and finds consensus
    communities of metagenes between studies which appear to represent
    similar metagenes (gene signatures), and tagging genes with their
    respective communities.">
<title>R Package for combining networks into a consensus network â€¢ consensusNetR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="R Package for combining networks into a consensus network">
<meta property="og:description" content="consensusNet provides tools for combining the results of
    metagene clustering in different studies, datasets, and using
    different methods.  It combines results and finds consensus
    communities of metagenes between studies which appear to represent
    similar metagenes (gene signatures), and tagging genes with their
    respective communities.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">consensusNetR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Systems-Methods/consensusNetR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="consensusnetr">consensusNetR<a class="anchor" aria-label="anchor" href="#consensusnetr"></a>
</h1></div>
<!-- badges: start -->

<p>consensusNetR is an R Package for combining networks into a consensus network based on the work of <a href="https://academic.oup.com/bioinformatics/article/35/21/4307/5426054" class="external-link">Laura Cantini and Andrei Zinovyev</a>. In addition to identifying consensus based on correlation of community meta-genes (loadings or membership scores), we also implement methods based on overlap.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install the version from BMS BioGit with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span></span>
<span>  repo <span class="op">=</span> <span class="st">"Systems-Methods/consensusNetR"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>or:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_git</span><span class="op">(</span></span>
<span>  <span class="st">"https://github.com/Systems-Methods/consensusNetR"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="example-workflow">Example Workflow<a class="anchor" aria-label="anchor" href="#example-workflow"></a>
</h1>
<p>This example will create a consensus network from three public datasets: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39582" class="external-link">GSE39582</a>, <a href="https://portal.gdc.cancer.gov/projects/TCGA-COAD" class="external-link">TCGA COAD</a>, and <a href="https://portal.gdc.cancer.gov/projects/TCGA-READ" class="external-link">TCGA READ</a></p>
<p>This example workflow begins with <a href="https://systems-methods.github.io/icWGCNA/" class="external-link">icWGCNA</a> results, however alternative methods work as well (KNN, PCA, ICA, WGCNA). See <a href="#example-appendix">Example Appendix</a> for downloading data and icWGCNA run details and code.</p>
<div class="section level2">
<h2 id="consensus-construction">Consensus construction<a class="anchor" aria-label="anchor" href="#consensus-construction"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create list of community_membership object</span></span>
<span><span class="va">memb_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  GSE39582 <span class="op">=</span> <span class="va">GSE39582_icwgcna</span><span class="op">$</span><span class="va">community_membership</span>,</span>
<span>  READ <span class="op">=</span> <span class="va">read_icwgcna</span><span class="op">$</span><span class="va">community_membership</span>,</span>
<span>  COAD <span class="op">=</span> <span class="va">coad_icwgcna</span><span class="op">$</span><span class="va">community_membership</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Construct Meta Reciprocal Best Hits based on overlaps (318 communities found)</span></span>
<span><span class="va">rbh</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/construct_rbh_overlap_based.html">construct_rbh_overlap_based</a></span><span class="op">(</span><span class="va">memb_list</span>, top_n <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rbh</span><span class="op">)</span></span>
<span><span class="co">## 318</span></span>
<span></span>
<span><span class="co"># RBH Heatmap Creation</span></span>
<span><span class="fu"><a href="reference/plot_rbh.html">plot_rbh</a></span><span class="op">(</span>rbh <span class="op">=</span> <span class="va">rbh</span>, network_membership_list <span class="op">=</span> <span class="va">memb_list</span><span class="op">)</span></span></code></pre></div>
<figure><img src="reference/figures/README-RBH-1.png" alt="Reciprocal Best Hits Heatmap"><figcaption aria-hidden="true">
Reciprocal Best Hits Heatmap
</figcaption></figure><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Detect Communities in Adjacency/Reciprocal Best Hits Matrix</span></span>
<span><span class="va">consensus_comms</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/detect_consensus_communities.html">detect_consensus_communities</a></span><span class="op">(</span><span class="va">rbh</span><span class="op">)</span></span>
<span><span class="co"># showing the first 10 communities </span></span>
<span><span class="co">#note community 1 is a miscellaneous and will be removed</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">consensus_comms</span><span class="op">$</span><span class="va">Cluster</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">##    1   2   3   4   5   6   7   8   9  10 </span></span>
<span><span class="co">##  200   3   3   3   3   3   3   3   3   3 </span></span>
<span></span>
<span><span class="co"># Compute the average metagene across studies for each community</span></span>
<span><span class="va">consensus_memb</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/calc_consensus_memberships.html">calc_consensus_memberships</a></span><span class="op">(</span><span class="va">consensus_comms</span>, <span class="va">memb_list</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="downstream-analysis">Downstream Analysis<a class="anchor" aria-label="anchor" href="#downstream-analysis"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">consensus_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_gene_community_membership.html">get_gene_community_membership</a></span><span class="op">(</span><span class="va">consensus_comms</span>, <span class="va">memb_list</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">consensus_genes</span><span class="op">)</span></span>
<span><span class="co">##    gene_id cluster n_studies</span></span>
<span><span class="co">##  1    A1BG      43         2</span></span>
<span><span class="co">##  2     A2M       1         2</span></span>
<span><span class="co">##  3    AACS      19         2</span></span>
<span><span class="co">##  4   AAGAB      25         2</span></span>
<span><span class="co">##  5   AASDH      40         2</span></span>
<span><span class="co">##  6    AASS      42         2</span></span>
<span></span>
<span><span class="co"># Need to use icWGCNA for individual eigengenes</span></span>
<span><span class="va">GSE39582_eigen</span> <span class="op">&lt;-</span> <span class="fu">icWGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/icWGCNA/man/compute_eigengene_matrix.html" class="external-link">compute_eigengene_matrix</a></span><span class="op">(</span></span>
<span>  ex <span class="op">=</span> <span class="va">GSE39582_df</span>,</span>
<span>  membership_matrix <span class="op">=</span> <span class="va">consensus_memb</span></span>
<span><span class="op">)</span></span>
<span><span class="va">read_eigen</span> <span class="op">&lt;-</span> <span class="fu">icWGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/icWGCNA/man/compute_eigengene_matrix.html" class="external-link">compute_eigengene_matrix</a></span><span class="op">(</span></span>
<span>  ex <span class="op">=</span> <span class="va">read_df</span>,</span>
<span>  membership_matrix <span class="op">=</span> <span class="va">consensus_memb</span></span>
<span><span class="op">)</span></span>
<span><span class="va">coad_eigen</span> <span class="op">&lt;-</span> <span class="fu">icWGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/icWGCNA/man/compute_eigengene_matrix.html" class="external-link">compute_eigengene_matrix</a></span><span class="op">(</span></span>
<span>  ex <span class="op">=</span> <span class="va">coad_df</span>,</span>
<span>  membership_matrix <span class="op">=</span> <span class="va">consensus_memb</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">eigen_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">GSE39582_eigen</span>, <span class="va">read_eigen</span>, <span class="va">coad_eigen</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/plot_consensus_eig_dist.html">plot_consensus_eig_dist</a></span><span class="op">(</span><span class="va">eigen_list</span><span class="op">)</span></span></code></pre></div>
<figure><img src="reference/figures/README-eig-dist-1.png" alt="Individual Eigengene Distributions"><figcaption aria-hidden="true">
Individual Eigengene Distributions
</figcaption></figure>
</div>
<div class="section level2">
<h2 id="example-appendix">Example Appendix<a class="anchor" aria-label="anchor" href="#example-appendix"></a>
</h2>
<details><summary>
Downloading data
</summary><p>For GSE39582 we need to convert from Affymetrix Human Genome U133 Plus 2.0 Array to gene symbols, by using the <code><a href="https://rdrr.io/pkg/icWGCNA/man/gene_mapping.html" class="external-link">icWGCNA::gene_mapping()</a></code> function. This matches with the two TCGA datasets already in gene symbols.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Systems-Methods/icWGCNA" class="external-link">icWGCNA</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># GSE39582</span></span>
<span><span class="va">GSE39582</span> <span class="op">&lt;-</span> <span class="fu">GEOquery</span><span class="fu">::</span><span class="fu"><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="st">"GSE39582"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># TCGA READ</span></span>
<span><span class="fu">UCSCXenaTools</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/UCSCXenaTools/reference/getTCGAdata.html" class="external-link">getTCGAdata</a></span><span class="op">(</span></span>
<span>  project <span class="op">=</span> <span class="st">"READ"</span>,</span>
<span>  mRNASeq <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mRNASeqType <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>  clinical <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  download <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  destdir <span class="op">=</span> <span class="st">"/MY_PATH/data/"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># TCGA COAD</span></span>
<span><span class="fu">UCSCXenaTools</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/UCSCXenaTools/reference/getTCGAdata.html" class="external-link">getTCGAdata</a></span><span class="op">(</span></span>
<span>  project <span class="op">=</span> <span class="st">"COAD"</span>,</span>
<span>  mRNASeq <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mRNASeqType <span class="op">=</span> <span class="st">"normalized"</span>,</span>
<span>  clinical <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  download <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  destdir <span class="op">=</span> <span class="st">"/MY_PATH/data/"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</details><details><summary>
Preprocessing steps
</summary><p>All datasets must have consistent annotation (i.e.Â Gene symbols, Entrez, Ensembl, â€¦). In this example we will convert GSE39582 to gene symbols using the <a href="https://systems-methods.github.io/icWGCNA/reference/gene_mapping.html" class="external-link">icWGCNA::gene_mapping()</a> function.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># creating annotation file for gene mapping to gene symbols</span></span>
<span><span class="va">GSE39582_annotation</span> <span class="op">&lt;-</span> <span class="va">GSE39582</span><span class="op">@</span><span class="va">featureData</span><span class="op">@</span><span class="va">data</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ID</span>, gene_symbol <span class="op">=</span> <span class="va">`Gene Symbol`</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    gene_symbol <span class="op">=</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>      <span class="va">gene_symbol</span>, <span class="op">~</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split</a></span><span class="op">(</span><span class="va">.x</span>, <span class="st">" /// "</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/unnest.html" class="external-link">unnest</a></span><span class="op">(</span><span class="va">gene_symbol</span><span class="op">)</span></span>
<span></span>
<span><span class="va">GSE39582_hugo</span> <span class="op">&lt;-</span> <span class="fu">icWGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/icWGCNA/man/gene_mapping.html" class="external-link">gene_mapping</a></span><span class="op">(</span></span>
<span>  <span class="va">GSE39582</span><span class="op">@</span><span class="va">assayData</span><span class="op">$</span><span class="va">exprs</span>,</span>
<span>  <span class="va">GSE39582_annotation</span>,</span>
<span>  compress_fun <span class="op">=</span> <span class="st">"highest_mean"</span>,</span>
<span>  compress_trans <span class="op">=</span> <span class="st">"log_exp"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>All data should be normalized, however in this example we downloaded the normalized data so no transformations needed here.</p>
</details><details><summary>
icWGCNA runs
</summary><p>For icWGCNA runs using defaults, except reducing max iterations to 5 for demonstration purposes. These runs benefit greatly by using multiple computer cores.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># GSE39582</span></span>
<span><span class="va">GSE39582_icwgcna</span> <span class="op">&lt;-</span> <span class="fu">icWGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/icWGCNA/man/icwgcna.html" class="external-link">icwgcna</a></span><span class="op">(</span><span class="va">GSE39582_hugo</span>, maxIt <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># TCGA READ</span></span>
<span><span class="va">read_icwgcna</span> <span class="op">&lt;-</span> <span class="fu">icWGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/icWGCNA/man/icwgcna.html" class="external-link">icwgcna</a></span><span class="op">(</span><span class="va">read_df</span>, maxIt <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># TCGA COAD</span></span>
<span><span class="va">coad_icwgcna</span> <span class="op">&lt;-</span> <span class="fu">icWGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/icWGCNA/man/icwgcna.html" class="external-link">icwgcna</a></span><span class="op">(</span><span class="va">coad_df</span>, maxIt <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
</details>
</div>
</div>
<div class="section level1">
<h1 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h1>
<p>Please note that the icWGCNA project is released with a <a href="https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/Systems-Methods/consensusNetR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/Systems-Methods/consensusNetR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing consensusNetR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://github.com/mjrmason" class="external-link">Mike Mason</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-5652-7739" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://github.com/wfulp" class="external-link">Jimmy Fulp</a> <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-6888-0794" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Thomas Corey <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-3044-2072" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://github.com/alexandre-alloy" class="external-link">Alexandre Alloy</a> <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-8280-0543" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://app.codecov.io/gh/Systems-Methods/consensusNetR?branch=main" class="external-link"><img src="https://codecov.io/gh/Systems-Methods/consensusNetR/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://github.com/mjrmason" class="external-link">Mike Mason</a>, <a href="https://github.com/wfulp" class="external-link">Jimmy Fulp</a>, Thomas Corey, <a href="https://github.com/alexandre-alloy" class="external-link">Alexandre Alloy</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
